jwst/stpipe/core.py

```diff
@@ -124,3 +137,14 @@ class JwstPipeline(Pipeline, JwstStep):
     def finalize_result(self, result, reference_files_used):
         if isinstance(result, JwstDataModel):
             log.info(f"Results used CRDS context: {crds_client.get_context_used(result.crds_observatory)}")
+
+            if hasattr(result, 'cal_logs'):
+                tmpdict = result.cal_logs.instance
+            else:
+                tmpdict = dict()
+            tmpdict[self.class_alias] = '\n'.join(self._log_records)
+
+            for _, step in self.step_defs.items():
+                if step.class_alias in tmpdict.keys():
+                    del tmpdict[step.class_alias]
+            result.cal_logs = tmpdict
```

Comment:
```suggestion
            if not hasattr(result, 'cal_logs'):
                result.cal_logs = {}
            
            for _, step in self.step_defs.items():
                if hasattr(result.cal_logs, step.class_alias):
                    delattr(result.cal_logs, step.class_alias)
                    
            setattr(result.cal_logs, self.class_alias, self._log_records)
```
Also switching `cal_logs` to a list of strings.